sql: TestRelocateNonVoters failed

sql.TestRelocateNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20265264?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20265264?buildTab=artifacts#/) on master @ [0e9251dba2fd91e153974e2701f73c93871f9d91](https://github.com/cockroachdb/cockroach/commits/0e9251dba2fd91e153974e2701f73c93871f9d91):


```
=== RUN   TestRelocateNonVoters
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/492cc08314ebb30515b6f60e383090f4/logTestRelocateNonVoters227813400
    test_log_scope.go:82: use -show-logs to present logs inline
    multitenant_admin_function_test.go:955: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/492cc08314ebb30515b6f60e383090f4/logTestRelocateNonVoters227813400
--- FAIL: TestRelocateNonVoters (73.51s)
=== RUN   TestRelocateNonVoters/ALTER_RANGE_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:374: condition failed to evaluate within 45s: from multitenant_admin_function_test.go:411: expected "ok" to be contained in result "trying to add({ChangeType:ADD_NON_VOTER Target:n1,s1}) to a store that already has a LEARNER" tenant=secondary query=``ALTER RANGE RELOCATE NONVOTERS FROM 5 TO 1 FOR (SELECT min(range_id) FROM [SHOW RANGES FROM TABLE t]);`` leaseholder=3 replicas=[2 3 4 5] voting_replicas=[2 3 4] non_voting_replicas=[5] fromReplica=5 toReplica=1 row=0 col=2
--- FAIL: TestRelocateNonVoters/ALTER_RANGE_RELOCATE_NONVOTERS (59.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #150469 sql: TestRelocateNonVoters failed [C-test-failure O-robot T-db-server branch-release-25.1.9-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53022

--- a/pkg/sql/BUILD.bazel
+++ b/pkg/sql/BUILD.bazel
@@ -810,6 +810,7 @@ go_test(
         "//pkg/kv/kvclient/rangefeed",
         "//pkg/kv/kvpb",
         "//pkg/kv/kvserver",
+        "//pkg/kv/kvserver/allocator",
         "//pkg/kv/kvserver/concurrency/isolation",
         "//pkg/kv/kvserver/kvserverbase",
         "//pkg/kv/kvserver/protectedts",

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -13,10 +13,12 @@ import (
 	"strconv"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/config/zonepb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/allocator"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/multitenant/tenantcapabilitiespb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
@@ -58,6 +60,13 @@ func createTestClusterArgs(ctx context.Context, numReplicas, numVoters int32) ba
 	clusterSettings := cluster.MakeTestingClusterSettings()
 	kvserverbase.LoadBasedRebalancingMode.Override(ctx, &clusterSettings.SV, kvserverbase.LBRebalancingOff)
 	kvserverbase.MergeQueueEnabled.Override(ctx, &clusterSettings.SV, false)
+
+	// Set allocator intervals to scan faster to help with recovery from race
+	// conditions between allocator and manual relocate operations.
+	allocator.LoadBasedRebalanceInterval.Override(ctx, &clusterSettings.SV, 100*time.Millisecond)
+	kvserver.MinLeaseTransferInterval.Override(ctx, &clusterSettings.SV, 100*time.Millisecond)
+	kvserver.MinIOOverloadLeaseShedInterval.Override(ctx, &clusterSettings.SV, 100*time.Millisecond)
+
 	return base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
 			Settings: clusterSettings,

